### PR TITLE
add one_hot encoding code, not implementend in pipeline yet

### PR DESCRIPTION
**Pull request by michellekappl**
implementation of one_hot encoding. Word level embedding is not supported

### CODE REVIEW

File: src/one_hot_encoding.jl [hunk 1]
```diff
@@ -1,3 +1,77 @@
...
+    julia> get_vocab(a)
+    4-element Array{String}:
+     "hello"
+     "world"
+     "second"
+     "sentence"
+    ```
+    """
+    vocab = []
+    for sentence in data
```
Comment (leonschreiber96) [inline src/one_hot_encoding.jl]: Just make a Set{String}

File: src/one_hot_encoding.jl [hunk 2]
```diff
@@ -1,3 +1,77 @@
-function one_hot(tokens)
-    #do something here
-end
\ No newline at end of file
+function one_hot(data, vocab = [])
```
Comment (leonschreiber96) [inline src/one_hot_encoding.jl]: Set{String}

File: src/one_hot_encoding.jl [hunk 3]
```diff
@@ -1,3 +1,77 @@
...
+    ```
+    """
+    if length(vocab) == 0
+        vocab = get_vocab(data)
+    end
+    vocab_dict = get_vocab_dict(vocab)
+    one_hot_data = zeros(Int, length(data), length(vocab))
+    for (i, sentence) in enumerate(data)
+        for word in sentence
+            one_hot_data[i, vocab_dict[word]] = 1
```
Comment (leonschreiber96) [inline src/one_hot_encoding.jl]: SparseMatrix?

**Review by leonschreiber96 — CHANGES_REQUESTED** [review]
Well done, please add some small changes.

**Comment by michellekappl** [comment]
code for the coming nlp pipeline is implemented, but must be uncommented

**Pull request merged**